scheduler/task.go

```diff
@@ -537,6 +535,37 @@ func (t *task) fire() {
 	t.state = core.TaskSpinning
 }
 
+// disable proceeds disabling a task which consists of changing task state to disabled, emitting an appropriate event
+// and unsubscribing its dependencies
+func (t *task) disable(failureMsg string) {
+	t.Lock()
+	t.state = core.TaskDisabled
+	t.Unlock()
+
+	// Send task disabled event
+	event := new(scheduler_event.TaskDisabledEvent)
+	event.TaskID = t.id
+	event.Why = fmt.Sprintf("Task disabled with error: %s", failureMsg)
+	defer t.eventEmitter.Emit(event)
+
+	// We need to unsubscribe from deps when a task has disabled
+	t.UnsubscribePlugins()
```

Comment:
@iza, sounds a good change. Only concern is that the disabled task can be enabled. Do we need to subscribe it before enabling it? Could you double check.